frontend/src/app/modules/plugins/plugin-context.ts

```diff
@@ -93,6 +95,10 @@ export class OpenProjectPluginContext {
     this.zone.run(cb);
   }
 
+  public registerAdditionalWorkPackageTab(tab: Tab) {
+    this.hooks.workPackageAdditionalTabs(() => tab)
+  }
```

Comment:
we could just use the plain old `this.hooks.workPackageAdditionalTabs` directly in plugins, but this new method makes using the hook type-safe. Thought this to be a little better than un-typed hooks.

Reply:
I would rather put this into the hook service itself rather than the plugin context. The latter was created for access into some angular services outside the angular application, such as within `CKEditor` or previously, within AngularJS 1 legacy compoents while we still ran both.

Reply:
cool, will attempt to move this to the hook service 🧑‍🏭